### PR TITLE
Add Naive Bayes and LDA model for benchmark

### PR DESCRIPTION
**Pull request by xukai92**
LDA runs super slow. I believe it's due to the large number of zs.

**TODO**
- [x] Apply tricks to speed up LDA.
- [x] Improve progress bar

### CODE REVIEW

**Comment by yebai** [comment]
The current progress bar looks a bit primitive 

```julia
1.0% 2.0% 3.0% 4.0% 5.0% 6.0% 7.0% 8.0% 9.0% 10.0% 11.0% 12.0% 13.0% 14.0% 15.0% 16.0% 17.0% 18.0% 19.0% 20.0% 21.0% 22.0% 23.0% 24.0% 25.0% 26.0% 27.0% 28.0% 29.0% 30.0% 31.0% 32.0% 33.0% 34.0% 35.0% 36.0% 37.0% 38.0% 39.0% 40.0% 41.0% 42.0% 43.0% 44.0% 45.0% 46.0% 47.0% 48.0% 49.0% 50.0% 51.0% 52.0% 53.0% 54.0% 55.0% 56.0% 57.0% 58.0% 59.0% 60.0% 61.0% 62.0% 63.0% 64.0% 65.0% 66.0% 67.0% 68.0% 69.0% 70.0% 71.0% 72.0% 73.0% 74.0% 75.0% 76.0% 77.0% 78.0% 79.0% 80.0% 81.0% 82.0% 83.0% 84.0% 85.0% 86.0% 87.0% 88.0% 89.0% 90.0% 91.0% 92.0% 93.0% 94.0% 95.0% 96.0% 97.0% 98.0% 99.0% 100% Done

```

A potentially better way to print progress is like

```julia
Progress: 53%[==========================>                       ]  ETA: 0:09:02
```

This can be done by using (or adapting) the [ProgressMeter](https://github.com/timholy/ProgressMeter.jl) package

**Comment by xukai92** [comment]
I will look into the progress bar this afternoon.

File: benchmarks/lda.model.jl [hunk 1]
```diff
@@ -37,11 +37,21 @@
...
+  map!(t -> Vector{Float64}(K), theta_p)
   for n = 1:N
-    z[n] ~ Categorical(theta[doc[n]])
+    theta_p[n][1] = phi[1][w[n]] * theta[doc[n]][1]
+    theta_p[n][2] = phi[2][w[n]] * theta[doc[n]][2]
+    theta_p[n] = theta_p[n] / sum(theta_p[n])
   end
+
+
+  z = Array{Int}(N)
```
Comment (yebai) [inline benchmarks/lda.model.jl]: we need to use `TArray` for `z` here.

Comment (xukai92) [inline benchmarks/lda.model.jl]: I see. Updated

**Comment by xukai92** [comment]
The benchmark job still got stopped at 49 min 58 sec.

**Pull request merged**